language-server/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -42,27 +46,49 @@ class ScalametaLanguageServer(cwd: AbsolutePath,
 
   override def onChangeWatchedFiles(changes: Seq[FileEvent]): Unit =
     changes.foreach {
-      case FileEvent(SemanticdbUri(path),
-                     FileChangeType.Created | FileChangeType.Changed) =>
+      case FileEvent(uri @ Uri(path),
+                     FileChangeType.Created | FileChangeType.Changed)
+          if uri.endsWith(".semanticdb") =>
         logger.info(s"$path changed or created. Running scalafix...")
         scalafixService.onSemanticdbPath(path).foreach(report)
       case event =>
         logger.info(s"Unhandled file event: $event")
         ()
     }
 
-  override def documentFormattingRequest(td: TextDocumentIdentifier,
-                                         options: FormattingOptions) = {
-    val path = new URI(td.uri)
-    val lines = scala.io.Source.fromFile(path).getLines.toList
-    val totalLines = lines.length
-    val fullDocumentRange = Range(
-      start = Position(0, 0),
-      end = Position(totalLines, lines.last.length)
-    )
-    val content = lines.mkString
-    val formattedContent = scalafmtService.formatDocument(content)
-    List(TextEdit(fullDocumentRange, formattedContent))
+  override def documentFormattingRequest(
+      td: TextDocumentIdentifier,
+      options: FormattingOptions): List[TextEdit] = {
+    try {
```

Comment:
it should be possible to make so that all exceptions are printed by modifying the LSP implementations. I'll take a look.